tests/integration/schema_registry/test_SchemaRegistryClient.py

```diff
@@ -0,0 +1,295 @@
+    """
+    Registers a schema, fetches that schema by it's subject version id.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the returned schema and id do not match the
+        registration schema and schema id.
+
+    """
+    sr = kafka_cluster.schema_registry()
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = _subject_name('test_registration')
+
+    schema_id = sr.register_schema(subject, schema)
+    registered_schema = sr.get_registration(subject, schema)
+
+    assert registered_schema.schema_id == schema_id
+    assert registered_schema.subject == subject
+
+
+def test_api_delete_subject_version(kafka_cluster, load_avsc):
+    """
+    Registers a Schema under a specific subject then deletes it.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the version number is not returned in the response or
+        if the subject has not been deleted.
+
+    """
+    sr = kafka_cluster.schema_registry()
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = str(uuid1())
+
+    sr.register_schema(subject, schema)
+    sr.delete_version(subject, 1)
+
+    assert subject not in sr.get_subjects()
+
+
+def test_api_subject_config_update(kafka_cluster, load_avsc):
+    """
+    Updates a subjects compatibility policy then ensures the same policy
+    is returned when queried.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the compatibility policy returned by the schema
+        registry fixture does not match the updated value.
+
+    """
+    sr = kafka_cluster.schema_registry()
+
+    schema = load_avsc('basic_schema.avsc')
+    subject = str(uuid1())
+
+    sr.register_schema(subject, schema)
+    sr.set_compatibility(subject_name=subject,
+                         level=CompatibilityType.FULL_TRANSITIVE)
+
+    assert sr.get_compatibility(
+        subject_name=subject)['compatibilityLevel'] == CompatibilityType.FULL_TRANSITIVE
+
+
+def test_api_config_update(kafka_cluster, load_avsc):
+    """
+    Updates a global compatibility policy then ensures the same policy
+    is returned when queried.
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): Kafka Cluster fixture
+        load_avsc (callable(str)): Schema fixture constructor
+
+    Raises:
+        AssertionError if the compatibility policy returned by the schema
+        registry fixture does not match the updated value.
+
+    """
+    sr = kafka_cluster.schema_registry()
+
+    sr.set_compatibility(level=CompatibilityType.FORWARD_TRANSITIVE)
+
+    assert sr.get_compatibility()['compatibilityLevel'] == CompatibilityType.FORWARD_TRANSITIVE
```

Comment:
This comment is for the entire file:
It is missing negative tests (i.e., doing operations on schemas that dont exist, invalid compat levels, etc).
Negative tests are as important as positive tests to make sure the error handling code works.